### DISCUSSION TITLE
Highlighted Arrows

### DISCUSSION THREAD

**Discussion by peyre** [raceintospace/raceintospace#616] [General]
I thought it would be a good idea to highlight up and down arrows so you could have an easy visual for whether you can scroll up and down, or not.  So I went and did all that, and only then realized that the game already had been doing that (though a bit more subtly) for left and right arrows.  It did it kind of the opposite way though: instead of highlighting arrows to show you _can_ scroll, it shows ones you can't as being depressed.  I wonder if it would be a good idea to remove my highlighted/regular arrows and make them work like these original regular/depressed buttons.  Anyone have any thoughts?

![1 horizontal buttons](https://user-images.githubusercontent.com/3118014/153770557-dd97076d-efaf-4193-b198-a7f2c54f11cf.png)
![2 vertical buttons](https://user-images.githubusercontent.com/3118014/153770559-f8bfc970-de80-469e-8841-eab3d3e07fe8.png)

**Comment by rnyoakum** [comment]
Highlighting/not highlighting the arrows does the job, but it's always bummed me out that the cost of that is losing the texture of the engraved arrow symbol. Depressing the buttons would symbolize they're locked out while restoring the nuance of the 'plain' arrow.

**Comment by peyre** [comment]
So, do you think it would be better if I went through and removed my white highlighted arrows and instead depressed arrows on buttons that were active?

That _would_ keep the esthetics of all the buttons consistent, though the depressed buttons are a bit subtle.

**Reply by rnyoakum** [reply]
I would, for consistency and preference, use the original arrow & depressed button scheme for indicating when scrolling is and is not available.

**Reply by peyre** [reply]
Thanks.  That should keep me busy tomorrow!

**Reply by peyre** [reply]
Man, this is a lot harder than I expected!  I did the arrow highlights by creating functions that lit up the arrow part of the button, but it turns out the Viewing Stand instead changes the borders with OutBox and InBox commands.  So far so good, but working on the newscast (start with something simple), it doesn't work to change the function calls to InBox/Outbox--the if statements don't work right for some reason.  I changed 
```
        if (ctop <= 0) {
            draw_up_arrow(305, 126);
        } else {
            draw_up_arrow_highlight(305, 126);
        }

        if (ctop >= bline) {
            draw_down_arrow(305, 163);
        } else {
            draw_down_arrow_highlight(305, 163);
        }
```

to 
```
        draw_up_arrow(305, 126);
        if (ctop > 0) {
            OutBox(301, 118, 315, 158);
        } else {
            InBox(301, 118, 315, 158);
        }

        draw_down_arrow(305, 163);
        if (ctop >= bline) {
            OutBox(301, 158, 315, 196);
        } else {
            InBox(301, 158, 315, 196);
        }
```

but the buttons don't raise and depress like they should.

**Reply by peyre** [reply]
Who, I got them working ok.  Now if only I could get them to look _right_.
![NewsButtons](https://user-images.githubusercontent.com/3118014/161393648-110369b4-3729-43ab-b61f-882fafc96e7c.png)

**Reply by peyre** [reply]
Well...maybe that's just how InBoxes and OutBoxes look.  I could do this in other places around, and we could always change the esthetics later...

**Reply by rnyoakum** [reply]
Nope, that's not how they look. If you make this change elsewhere, it'll just be more work undoing it. I'll try and see how it's done in other locations (such as the viewing stand). My guess is that the dimensions for the call to InBox and OutBox are inside the button outlines by a pixel or two.

**Reply by peyre** [reply]
Shucks.  Well, I got it working right somehow in Basic/Advanced Training, by using (in ast3.cpp):
```
    InBox(6, 130, 18, 161);  // Depressed button

    if (count > 8) {
        InBox(6, 130, 18, 161);  // Depressed button
    }
...
            if ((now2 < count - 1 && count > 8 && (8 - BarA) + now2 < count) || BarA == 0) {
                InBox(6, 130, 18, 161);  // Depressed button
            } else {
                OutBox(6, 130, 18, 161);  // Active button
            }
            if (now2 == 0) {
                InBox(6, 130, 18, 161);  // Depressed button
            } else {
                OutBox(6, 130, 18, 161);  // Active button
            }
            if (now2 == count - 1) {
                InBox(6, 163, 18, 194);  // Depressed button
            } else {
                OutBox(6, 163, 18, 194);  // Active button
            }
```

![Basic](https://user-images.githubusercontent.com/3118014/161403211-bdadf937-b7e7-4064-a08f-b4ac83df4cc6.png)

**Reply by peyre** [reply]
The Time Capsule is giving a lot of trouble too, like the newscast.  But as you suggested, I'll hold off making more changes (I haven't uploaded any yet either).

**Comment by peyre** [comment]
FWIW, the highlighted arrows are defined in draw.h and draw.cpp, and those lines should probably be removed if we manage to implement all this.  Pages that do the highlighting are:
news.cpp
ast3.cpp
place.cpp
admin.cpp
intel.cpp
budget.cpp
newmis.cpp

**Reply by peyre** [reply]
I thought maybe the margins on the buttons in, say, the Viewing Stand were big enough, while those in the newscast didn't have room to show the OutBox and InBox the way we wanted, but no, the margins look about the same.  Maybe it's that the OutBox/Inbox thing just doesn't look light in vertical boxes the way it does in horizontal ones?

**Reply by peyre** [reply]
Oh wait, I got it working in Basic Training.  I wonder what's different about the buttons in the newscast!
